### PR TITLE
[Ops][Feature] Add Triton kernel warmup for Ascend NPU

### PR DESCRIPTION
**Pull request by keyi-zz**
### What this PR does / why we need it?
This PR introduces Triton kernel warmup functionality for Ascend NPU to JIT-compile kernels (rejection sampler, penalties, and RMS norm) before actual model execution, preventing latency spikes during the first inference steps. It also updates various Triton kernels to use `do_not_specialize` for dynamic parameters to avoid excessive recompilation.

Feedback:
- In `kernel_warmup.py`, the imported `dsa_triton_warmup` is not called and should be added to the warmup sequence.
- In `penalties_triton_warmup.py` and `rms_triton_warmup.py`, accessing `worker.model_config` directly will raise an `AttributeError` and should be updated to `worker.vllm_config.model_config`.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
rejection_sampler Triton warmup complete in 31.635s.
penalties Triton warmup complete in 6.285s.
rms Triton warmup complete in 14.759.
Triton kernel warmup finished in 52.639s.

./Yr8IINFq5VRkjFlDMMxyfj0tDPedD9iugg3hsnXXzDo/prepare_inputs_padded_kernel.ttir
./DwwQgipSNvXn6B_v4_LEeuQCQHPIrMQfT--PeO70mqk/expand_kernel.ttir
./T6AYHvrUOJw5lj7oW2F31yW0X33wvboKk379V2ATNeg/expand_kernel.ttir
./B6Hf-RQrtEwZ-laWMfuxt5R982giBJKlUGvgmPifRTw/expand_kernel.ttir
./_Pk_bVCq4IAUzXhbXjMfPDEoxCbZnjldM77OGW9KmkU/expand_kernel.ttir
./yfIOPQdc512Qmxa2w1TmqTIhPX612s7hKmO21mJbKes/rejection_random_sample_kernel.ttir
./emp_IPKL5g2uGwZ4lV-Ao4Pr_l8rng15OQdkMjnwu_8/rejection_random_sample_kernel.ttir
./bQMnOYFKAnT5tpz3b2jAaN9TIqMZjyQsgFhUj6Inszc/sample_recovered_tokens_kernel.ttir
./aUtadCusPwaScrL7X8Lnb25k-atiBv1i9nj-nU80Ka0/rejection_greedy_sample_triton.ttir
./RMG_hIIPbwAWwaCp_syAVdViaeNmGuh7Dt5veG4grlw/rejection_greedy_sample_triton.ttir
./bJUueJV7ohaT7umopA5h6AEMU_e7waV5OG7Dq5Z476Y/rejection_greedy_sample_spec_len_1_triton.ttir
./QQURF9I6uDWqmWYAUhDObelnkYi0a9S3tq9A_gNzOG0/rejection_greedy_sample_spec_len_1_triton.ttir
./l1cCQGqG8K90NY4WPpJTFd1RVXHh8PYxP9h-6dzn0u0/triton_rms_kernel.ttir
./t78zEGk3VvreqJKLPYH8UkcV1AQ7o01GqtVwr7RBUDQ/triton_rms_kernel.ttir
./TjbaTa7Hql7ECHgSC_8MjSVZ0PWr3MbVUft_ICXaPes/triton_rms_kernel.ttir
./uzRoq5DcyNtv8rpwMsB66yV_-kkRKxDOvmM5l-2E7aI/triton_rms_kernel.ttir
./KQoCNndAa39eaoGxa11xaKLarwK93AFIZwDA4cDCDFY/triton_rms_kernel.ttir
./e3ARyJa8vqme8K7ZGabGrRM80-g1jg_D563AvKJQCu8/token_bin_counts_and_mask_kernel.ttir

- vLLM version: v0.27.1
- vLLM main: https://github.com/vllm-project/vllm/commit/58d3918e3ea0a544ffedadad2ba84559e9c51d8f

### CODE REVIEW

File: vllm_ascend/ops/triton/penalty.py [hunk 1]
```diff
@@ -155,4 +163,5 @@ def _apply_all_penalties_triton(
         stride_bin_counts_seq=output_bin_counts.stride(0),
         stride_bin_counts_vocab=output_bin_counts.stride(1),
         BLOCK_SIZE=2048,
+        multibuffer=False,
```
Comment (realliujiaxu) [inline vllm_ascend/ops/triton/penalty.py]: Is this necessary?

File: vllm_ascend/ops/triton/penalty.py [hunk 2]
```diff
@@ -27,7 +27,15 @@
 from vllm_ascend.ops.triton.triton_utils import get_vectorcore_num
 
 
-@triton.jit
+@triton.jit(
+    do_not_specialize=[
+        "num_seqs",
+        "stride_logits_seq",
```
Comment (realliujiaxu) [inline vllm_ascend/ops/triton/penalty.py]: I think stride will not change during the whore serving process, please check.

**Review by ZT-AIA — APPROVED** [review]

**Review by chenlongxiao — APPROVED** [review]

**Review by yiz-liu — APPROVED** [review]
